/mips/sgemv_n_msa.c:426:33: error: incompatible types when initializing type ‘v4f32’ using type ‘double’

HI :
   i compile the openblas , using the doc:
MTI_TOOLCHAIN_DIR={/home/zsf/WorkSpace/Kernel/JunZheng/mips-mti-linux-gnu/2016.05-03/bin}
MTI_GCC_PREFIX=mips-mti-linux-gnu
MTI_TOOLCHAIN=${IMG_TOOLCHAIN_DIR}/${IMG_GCC_PREFIX}

make BINARY=32 BINARY32=1 CC=$MTI_TOOLCHAIN-gcc AR=$MTI_TOOLCHAIN-ar FC="$MTI_TOOLCHAIN-gfortran -EL"    RANLIB=$MTI_TOOLCHAIN-ranlib HOSTCC=gcc CFLAGS="-EL" FFLAGS=$CFLAGS LDFLAGS=$CFLAGS TARGET=P5600


but have the error :+1: 
ITY -I.. -UDOUBLE  -UCOMPLEX -UDOUBLE -UCOMPLEX  -DTRANS  ../kernel/mips/../mips/sgemv_t_msa.c -o sgemv_t.o
../kernel/mips/../mips/sgemv_n_msa.c: In function ‘sgemv_n’:
../kernel/mips/../mips/sgemv_n_msa.c:426:33: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 v_alpha, x0, x1, y0 = 0.0, y1 = 0.0;
                                 ^
../kernel/mips/../mips/sgemv_n_msa.c:426:43: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 v_alpha, x0, x1, y0 = 0.0, y1 = 0.0;
                                           ^
../kernel/mips/../mips/sgemv_n_msa.c:428:17: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                 ^
../kernel/mips/../mips/sgemv_n_msa.c:428:28: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                            ^
../kernel/mips/../mips/sgemv_n_msa.c:428:39: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                       ^
../kernel/mips/../mips/sgemv_n_msa.c:428:50: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                                  ^
../kernel/mips/../mips/sgemv_n_msa.c:428:61: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                                             ^
../kernel/mips/../mips/sgemv_n_msa.c:428:72: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                                                        ^
../kernel/mips/../mips/sgemv_n_msa.c:428:83: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                                                                   ^
../kernel/mips/../mips/sgemv_n_msa.c:428:94: error: incompatible types when initializing type ‘v4f32’ using type ‘double’
     v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
                                                                                              ^
make[1]: *** [sgemv_n.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/home/zsf/WorkSpace/Kernel/JunZheng/OpenBLAS/kernel'
make: *** [libs] Error 1

--- a/kernel/mips/dgemv_n_msa.c
+++ b/kernel/mips/dgemv_n_msa.c
@@ -484,10 +484,10 @@ int CNAME(BLASLONG m, BLASLONG n, BLASLONG dummy1, FLOAT alpha, FLOAT *A,
     FLOAT *pa0, *pa1, *pa2, *pa3, *pa4, *pa5, *pa6, *pa7;
     FLOAT temp, temp0, temp1, temp2, temp3, temp4, temp5, temp6, temp7;
     v2f64 v_alpha;
-    v2f64 x0, x1, x2, x3, y0 = 0.0, y1 = 0.0, y2 = 0.0, y3 = 0.0;
+    v2f64 x0, x1, x2, x3, y0 = {0,0}, y1 = {0,0}, y2 = {0,0}, y3 = {0,0};
     v2f64 t0, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, t11, t12, t13, t14, t15;
     v2f64 t16, t17, t18, t19, t20, t21, t22, t23, t24, t25, t26, t27, t28, t29;
-    v2f64 t30, t31, tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
+    v2f64 t30, t31, tp0 = {0,0}, tp1 = {0,0}, tp2 = {0,0}, tp3 = {0,0}, tp4 = {0,0}, tp5 = {0,0}, tp6 = {0,0}, tp7 = {0,0};
 
     v_alpha = COPY_DOUBLE_TO_VECTOR(alpha);
 

--- a/kernel/mips/sgemv_n_msa.c
+++ b/kernel/mips/sgemv_n_msa.c
@@ -423,9 +423,9 @@ int CNAME(BLASLONG m, BLASLONG n, BLASLONG dummy1, FLOAT alpha, FLOAT *A,
     FLOAT *y_org = y;
     FLOAT *pa0, *pa1, *pa2, *pa3, *pa4, *pa5, *pa6, *pa7;
     FLOAT temp, temp0, temp1, temp2, temp3, temp4, temp5, temp6, temp7;
-    v4f32 v_alpha, x0, x1, y0 = 0.0, y1 = 0.0;
+    v4f32 v_alpha, x0, x1, y0 = {0,0,0,0}, y1 = {0,0,0,0};
     v4f32 t0, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, t11, t12, t13, t14, t15;
-    v4f32 tp0 = 0.0, tp1 = 0.0, tp2 = 0.0, tp3 = 0.0, tp4 = 0.0, tp5 = 0.0, tp6 = 0.0, tp7 = 0.0;
+    v4f32 tp0 = {0,0,0,0}, tp1 = {0,0,0,0}, tp2 = {0,0,0,0}, tp3 = {0,0,0,0}, tp4 = {0,0,0,0}, tp5 = {0,0,0,0}, tp6 = {0,0,0,0}, tp7 = {0,0,0,0};
 
     v_alpha = COPY_FLOAT_TO_VECTOR(alpha);
 

